apm-agent-core/src/main/java/co/elastic/apm/agent/report/AbstractIntakeApiHandler.java

```diff
@@ -200,25 +211,22 @@ protected boolean isApiRequestTimeExpired() {
         return System.nanoTime() >= requestStartedNanos + TimeUnit.MILLISECONDS.toNanos(reporterConfiguration.getApiRequestTime().getMillis());
     }
 
-    protected void onRequestError(Integer responseCode, InputStream inputStream, @Nullable IOException e) {
-        // TODO read accepted, dropped and invalid
-        onConnectionError(responseCode, currentlyTransmitting, 0);
+    private void onRequestError(Integer responseCode, long bytesWritten, InputStream inputStream, @Nullable IOException e) {
+        String responseBody = null;
+        try {
```

Comment:
is there a reason for removing the `isWarnEnabled` guard?

Reply:
Yes, previously the `isWarnEnabled` guard was there because we only parsed the APM server response body in case the log level was `WARN`.
Now we always parse the response independently of the log level, because we read the `accepted` field from it.
Therefore the only thing which would be left within the guard would be the `log.warn`s, which make the guard pointless.